examples/pipelines/fl_cross_silo_factory/submit.py

```diff
@@ -107,13 +128,21 @@
         "Could not find config.json, using config.yaml refs to Azure ML workspace instead."
     )
 
-    # tries to connect using provided references in config.yaml
-    ML_CLIENT = MLClient(
-        subscription_id=YAML_CONFIG.aml.subscription_id,
-        resource_group_name=YAML_CONFIG.aml.resource_group_name,
-        workspace_name=YAML_CONFIG.aml.workspace_name,
-        credential=credential,
-    )
+    # tries to connect using cli args if provided else using config.yaml
+    if args.subscription_id and args.resource_group_name and args.workspace_name:
+        ML_CLIENT = MLClient(
+            subscription_id=args.subscription_id,
```

Comment:
you can also set default value for these args to be None, then use `args.subscription_id or YAML_CONFIG.aml.subscription_id`

this would make it a little lighter

Reply:
The default value is already managed by the `argparse`.

Reply:
Changed the way though